File: Python/number-of-even-and-odd-bits.py
# Time:  O(1)
# Space: O(1)

# bit manipulation
class Solution(object):
    def evenOddBit(self, n):
        """
        :type n: int
        :rtype: List[int]
        """
        def popcount(x):
            return bin(x)[2:].count('1')

        return [popcount(n&0b0101010101), popcount(n&0b1010101010)]

File: C++/number-of-even-and-odd-bits.cpp
// Time:  O(1)
// Space: O(1)

// bit manipulation
class Solution {
public:
    vector<int> evenOddBit(int n) {
        return {__builtin_popcount(n & 0b0101010101),
                __builtin_popcount(n & 0b1010101010)};
    }
};
